src/gsi/gsimod.F90

```diff
@@ -1821,7 +1822,8 @@ subroutine gsimain_initialize
     else 
        write(6,*)'n_ens_gfs and n_ens_fv3sar won"t be used if not regional_ensemble_option==5' 
     endif
-    
+    weight_ens_gfs=one
```

Comment:
Here, the namelist parameters like l_both_fv3sar_gfs_ens had been readin as well as weight_ens_gfs and weight_ens_fv3sar, right?  then those assignments of the later two parameters (weight_ens??)  will override the readin values for them. So, they will always be one. 
Do I misunderstand something here?

Reply:
In the case of l_both_fv3sar_gfs_ens=.false., only either global or regional ensemble is used, so weight_ens_gfs and weight_ens_fv3sar should be one (only either is used). To force that, I added this part. (If ensemble B should be tuned in this case, please don't set weight_ens_gfs or weight_ens_fv3sar but set beta_e0.) In the case of l_both_fv3sar_gfs_ens=.true., weight_ens_gfs and weight_ens_fv3sar are not overwritten and the parameters in the namelist are used.

Reply:
@shoyokota Thanks for your explanation . That does make sense. 
This question is resolved

Reply:
@shoyokota Could you please run a regression test and document the test result here.

Reply:
@ShunLiu-NOAA I tried regression tests on Orion, but global 3dvar/4denvar/4dvar tests were failed because accesses to some files belonging to group "rstprod" are not permitted. I requested access to the NOAA RDHPCS Program, but it has not been accepted yet after two weeks. Even after it is accepted, I can run tests only on Orion. Could I ask someone to run the regression tests on my behalf?

Reply:
@TingLei-NOAA Could you please run a regression test on HERA or WCOSS2?

Reply:
@ShunLiu-NOAA  Sure, I will give an update here when I have finished the regression tests.